git-repository/examples/init-non-bare-repo.rs

```diff
@@ -0,0 +1,33 @@
+// cargo run -p git-repository --example new
+
+use anyhow::Context;
+use git_repository as git;
+
+fn main() -> anyhow::Result<()> {
+    let work_dir = std::env::args_os()
+        .nth(1)
+        .context("First argument needs to be the directory to initialize the repository in")?;
+    let repo = git::init(work_dir)?;
+
+    println!(
+        "Repo: {:?}",
+        repo.work_dir().expect("non-bare repositories have a work-dir")
```

Comment:
by the way, I have been using git for about a decade (tho never with an API) and have no idea what a "bare" or "non-bare" repository is.  Maybe consider comment at the top with description of what the example does (and maybe explain that in context)... or a name like `init-repo-and-commit.rs`

Reply:
That's fair! Please feel free to rename it, I like `init-repo-and-commit.rs` more as well as it's keeping things simple for the common case that most people know. There might be another example one day that deals with a bare repository, and that one could get some hints at what the difference is.